pulser-core/pulser/sequence/sequence.py

```diff
@@ -370,6 +370,82 @@ def config_slm_mask(self, qubits: Iterable[QubitId]) -> None:
         # If checks have passed, set the SLM mask targets
         self._slm_mask_targets = targets
 
+    def switch_device(
+        self, new_device: Device, strict: bool = False
+    ) -> Sequence:
+        """Switch the device of a sequence.
+
+        Args:
+            new_device: The target device instance
+            strict: Add matching requirements making the switch more convenient.
+        """
+        # Check if the device is new or not
+
+        if self._device == new_device:
+            warn("Watchout, you're switching with the same device",stacklevel=2)
+            return self
+        if self._device == MockDevice:
+            raise ValueError("Switches from MockDevice are not allowed.")
+        # Initialize the new sequence
+        new_seq = Sequence(self.register, new_device)
+
+        # Recall the calls
+        called = False
+        strict_error_message = None
+        ch_type_er_mess = None
+        for call in self._calls[1:]:
+            if call.name == "declare_channel":
+                od_ch_obj = self._schedule[call.args[0]].channel_obj
+                for nd_ch_id, nd_ch_obj in new_seq.available_channels.items():
+                    # Find the corresponding channel on the new device
+                    # We verify the channel class then
+                    # check whether it's Global or local
+                    basis_match = od_ch_obj.basis == nd_ch_obj.basis
+                    addressing_match = (
+                        od_ch_obj.addressing == nd_ch_obj.addressing
+                    )
+                    if basis_match and addressing_match:
+                        if strict:
+                            phase_jump_time_check = (
+                                od_ch_obj.phase_jump_time
+                                != nd_ch_obj.phase_jump_time
+                            )
+                            clock_period_check = (
+                                od_ch_obj.clock_period
+                                != nd_ch_obj.clock_period
+                            )
```

Comment:
Thinking on this more deeply, these checks don't have to be so strict. It's a bit more involved but I think it's worth it:
- The clock period is only invalid if the old device clock period is not a multiple of the new device clock period. So for example, if you're changing from 4ns, clock periods of 1ns and 2ns would be valid, everything else would not. This is because the clock period enforces that every duration is a multiple of it, but a multiple of 4 is also a multiple of 1 and 2 so all elements in the sequence will still keep their duration unchanged. Note, however, that the inverse is not true: changing from 1ns to 4ns should fail.
- The phase jump time is only relevant when there are phase jumps in the sequence (ie consecutive pulses where the phase changes). An easy check you can do here is to see if the phase values as given by the [sampler](https://pulser.readthedocs.io/en/stable/apidoc/core.html#module-pulser.sampler.sampler) are constant - if they are, the phase jump time is irrelevant.